tests/D2L.CodeStyle.Analyzers.Test/Specs/DependencyRegistrationsAnalyzer.cs

```diff
@@ -232,42 +211,43 @@ public static class SomeExtensionMethods {
 
 		public static class OnlyRegistrationMethodsOnIDependencyRegistryMatter {
 			public static void DoesntMatter( IDependencyRegistry reg ) {
-				Register<UnmarkedSingleton>( null );
+				Register<INotSingleton>( null );
 			}
 			public static void Register<TDependencyType>(
 				TDependencyType instance
 			);
 		}
 	}
 
-	// intentionally unmarked
-	public interface ISingleton {}
-
 	[Singleton]
-	public interface IMarkedSingleton : ISingleton { }
-	public sealed class MarkedSingleton : IMarkedSingleton { }
+	public interface ISingleton { }
+	public interface INotSingleton { }
+
+	[Objects.Immutable]
+	public interface IImmutableSingleton : ISingleton { }
 
-	public interface IUnmarkedSingleton { }
-	public interface ISubUnmarkedSingleton : IUnmarkedSingleton { }
-	public sealed class UnmarkedSingleton : IUnmarkedSingleton {}
+	public interface IImmutableSubSingleton : IImmutableSingleton { }
+
+	[Objects.Immutable]
+	public sealed class ImmutableThing : ISingleton, INotSingleton { }
 
 	public interface ICreatedByDynamicFactory { }
 
 	public class ThingThatIsCreatedByDynamicObjectFactoryViaMarkedThing : ICreatedByDynamicFactory {
 		public ThingThatIsCreatedByDynamicObjectFactoryViaMarkedThing(

```

Comment:
You changed the other names in this class to use "Immutable" instead of "Marked"/"Unmarked".  Similarly, we might want to change this to `CreatedByDynamicFactoryWithImmutableDependency" or something like that

Reply:
Yes, I'll do that now.

Reply:
Done.